Added `nth` to `BitVec::Iter` so that `.skip(n)` gets optimized

hacatu opened the pull request:
Fixes #122

pczarn commented:
We could have it documented that `nth`/`skip` is O(n) so nobody wonders about the reason this code is necessary (which is time complexity)

pczarn commented:
Could use a benchmark as well

hacatu commented:
Ok, I added a benchmark, basically create a bitvec where we skip the first `3 << 20` entries and read the last `16411` (much smaller) entries within the benchmark.  This confirms what we expect:
before:
```
1,153,510.30 ns/iter (+/- 2,229.89)
```
after:
```
6,091.13 ns/iter (+/- 142.41)
```

I added an explanation that `Iterator::skip` will use `Iterator::nth` to the `nth` override, let me know if I should expand it

pczarn commented on src/lib.rs:
```diff
@@ -1937,6 +1937,13 @@ impl<'a, B: BitBlock> Iterator for Iter<'a, B> {
+        // This override is used by the compiler to optimize Iterator::skip.
+        // Without this, the default implementation of Iterator::nth is used, which walks over
+        // the whole iterator up to n.
+        self.range.nth(n).and_then(|i|self.bit_vec.get(i))
```
This will not pass fmt due to lack of whitespace

pczarn replied:
Please run `cargo fmt`

pczarn commented:
Thanks a lot!
